### PR TITLE
chore: add AutomateTaskCreatorUpgradeable & AutomateReadyUpgradeable

### PR DESCRIPTION
**Pull request by brandonchuah**

### CODE REVIEW

**Review by donoso-eth — COMMENTED** [review]
@brandonchuah in line 37 there is a constructor, if this contract will be inherit by a upgradeable contract, it won't be a constructor call, right?

**Comment by brandonchuah** [comment]
@donoso-eth the constructor will be used to initialize the`automate` address which is immutable.

File: contracts/integrations/AutomateReadyUpgradeable.sol [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+     * Only tasks created by _taskCreator defined in constructor can call
+     * the functions with this modifier.
+     */
+    modifier onlyDedicatedMsgSender() {
+        require(msg.sender == dedicatedMsgSender, "Only dedicated msg.sender");
+        _;
+    }
+
+    constructor(address _automate) {
+        automate = IAutomate(_automate);
```
Comment (goums) [inline contracts/integrations/AutomateReadyUpgradeable.sol:34]: One question @brandonchuah , if someone has a live upgradable contract, which is not inheriting `AutomateReadyUpgrdable` yet, and want to upgrade it to extend `AutomateReadyUpgradable`.
Then this constructor won't be called?

Comment (brandonchuah) [reply contracts/integrations/AutomateReadyUpgradeable.sol:34]: Usually, the upgradable contract templates are used for new undeployed contracts, it is not advisable to inherit `AutomateReadyUpgrdable` for an already deployed contract as there will be potential clashing of storage slots. A more customized approach is recommended. 

But this constructor will still be called by the parent contract when the upgradable contract is redeployed.

Comment (goums) [reply contracts/integrations/AutomateReadyUpgradeable.sol:34]: Ok, thanks for the info @brandonchuah

**Pull request merged**